src/wallet/rpc/spend.cpp

```diff
@@ -1167,7 +1167,7 @@ RPCHelpMan walletprocesspsbt()
                 {
                     {"psbt", RPCArg::Type::STR, RPCArg::Optional::NO, "The transaction base64 string"},
                     {"sign", RPCArg::Type::BOOL, RPCArg::Default{true}, "Also sign the transaction when updating (requires wallet to be unlocked)"},
-                    {"sighashtype", RPCArg::Type::STR, RPCArg::Default{"DEFAULT"}, "The signature hash type to sign with if not specified by the PSBT. Must be one of\n"
+                    {"sighashtype", RPCArg::Type::STR, RPCArg::Default{"DEFAULT for Taproot, ALL otherwise"}, "The signature hash type to sign with if not specified by the PSBT. Must be one of\n"
```

Comment:
How is it "ALL otherwise"? If a user wants `ALL`, even for non-Taproot descriptor, they now have to explicitly set it. It's a (minor) breaking change which i think should have had a release note.

More confusing is that even if `ALL` is set in the PSBT input, this will use `DEFAULT`. A user has to specify it using the command line for the whole transaction.

Reply:
`ALL` is the sighash type that is signed with by default for non-Taproot things. I don't see how this is incorrect?

Reply:
Hmm? That's the point. It's not.

You changed `ParseSighashString` in this PR setting `DEFAULT` for all types, not only Taproot.
https://github.com/bitcoin/bitcoin/blob/19526d937fbd16b3c868f4a7ecd2c076e71eb8fc/src/core_read.cpp#L255-L277

Reply:
`DEFAULT` is an alias for `ALL` internally, and signing non-taproot things will result in a `ALL` being appended to the signature.

Reply:
Right, i think the issue was because in my repro i had the sighash flag set in the PSBT input. Have i tried to reproduce it before reporting i could have been more precise, sorry about that.

It's still a (minor) API break since the PSBT was signed without failure in 22.0 but failed in 23.0 (as shown by the necessity to modify the functional test for it to pass in this PR https://github.com/bitcoin/bitcoin/pull/22514/files#r754474555). But since it was released like this already i agree with your comment in #25876 that this doesn't need "fixing" (not sure there could have been any reasonable fix anyways).